core/model/src/driver.rs

```diff
@@ -298,6 +298,13 @@ impl Exit {
             token,
         }
     }
+
+    pub fn amount(&self) -> Option<BigDecimal> {
+        match &self.amount {
+            Some(a) => Some(a.clone()),
```

Comment:
`BigDecimal` does not clone as primitive types which makes me trouble... is it OK, alternatives :question: 
:bow:

Reply:
This is OK, but I think you can simplify the code by simply calling `self.amount.clone()` and get rid of the `match`.